### PR TITLE
ci: only generate preview on collaborator comment

### PR DESCRIPTION
**Pull request by ROpdebee**
Collaborators and above must create a comment starting with `/deploy-preview`
to generate a preview for the current PR. This is because PRs
originating from forked repos (incl. dependabot) don't have write
access to push to a comparison branch.

@kellnerd: Pinging you since you're the only collaborator on this repo at the moment. Basically, these changes _should_ us to create a preview of the changes made in a PR by adding a comment where the first line is `/deploy-preview`. The bot should now leave a comment saying whether or not changes to the built userscript will occur, but it won't generate a built preview that can be installed or compared to the current version yet. The main reason is because it's a bit unsafe: Someone could change the build scripts to do malicious things and the bot would happily execute all of that... So that's disabled, and you have to explicitly "run" that `/preview` "command" (but only if it's safe, of course 😄 )

Edit: Of course, the bot also doesn't have the permissions to leave comments on PRs from forks, and I think it's a bit overkill to try and fix that, so I'll make another change to not do the check automatically either.

TL;DR: If you need to preview changes to the built userscripts in a PR, comment `/deploy-preview` and the bot will do it.

### CODE REVIEW

**Comment by ROpdebee** [comment]
Let's not talk about the mess of the last few days 😂 Absolutely horrible.

Quick summary of the horror show of setting this up:
1. Initially, I wanted the bot to comment a summary of the changes to built userscripts, with a preview installable version  and a diff. That worked on my PR 🎉 but failed on dependabot's PRs, because on those PRs (and any PR from forks), CI has no write permissions to the repo, for security purposes.
2. I changed it to have the bot _check_ whether changes would be made, and comment on the PR about it, but not push a preview version (so no diffs or installable version) automatically. The idea would be that the `/preview` command could be used to trigger that preview. That was this PR.
3. That _still_ didn't work, because CI didn't have permissions to comment on forked PRs either, so I decided to put all of that stuff behind the `/preview` command. I got a bit fed up with the whole thing, so I disabled the branch protection rules and started pushing directly to `main`.
4. Then I noticed that the `/preview` command didn't work, it was named `/deploy-preview` instead, and I was too sick of this shit to try to fix that again, so `/deploy-preview` it'll be 😄 

Basically: If you comment `/deploy-preview` on a PR now, the bot will tell you what changes will be made to the released userscripts! Hooray 🎉

I'm not planning on doing much work on the workflows anymore in the near future. You can start watching the repo again, if you'd like 😛

**Comment by kellnerd** [comment]
> You can start watching the repo again, if you'd like 😛

You got me 😂 Unfortunately I had only stopped watching the repo *after* I had all of today's Github bot spam already in my inbox. I thought the email filters, which I had setup yesterday, would spare me from that, but Thunderbird spent like five minutes to download them nevertheless just to let the filters delete them almost immediately 😬
Why doesn't GitHub offer any options to exclude comments by bots? I only had the choice between getting all notifications for all issues and PRs (including those by dependabot) or none of them, but I found no way to reduce bot spam on GitHub's side.

> Honestly though, I've also disabled most of the e-mail notifications, I felt like my head was about to explode.

Did you have more success than me to find the right option or have you just temporarily disabled all of them? In the GitHub settings or only for this repository?

> I got a bit fed up with the whole thing, so I disabled the branch protection rules and started pushing directly to main.

Ah, so that's why I missed that the overseen /preview command is no longer relevant, I have only checked the PRs (after I had cleaned my inbox from like 70 dependabot mails from the last 24 hours to review the good ones 🤯).
